### PR TITLE
Nrf52 & SDK 11

### PR DESCRIPTION
**Pull request by mrquincle**
@trond-snekvik This supports nRF52 and SDK 11. It introduces the macros `NRF51` and `NORDIC_SDK_VERSION >= 11`.

### CODE REVIEW

**Comment by trond-snekvik** [comment]
This is fantastic! :+1: 

We'll spend some time to work through the implication of the changes, and see if we have some restructuring to do (the nRF51 folder should probably be renamed, for instance). 

Great work!

File: nRF51/rbc_mesh/src/mesh_aci.c [hunk 1]
```diff
@@ -119,7 +126,11 @@ static void serial_command_handler(serial_cmd_t* p_serial_cmd)
 #ifdef SOFTDEVICE_PRESENT
             sd_power_reset_reason_clr(0x0F000F);
             sd_power_gpregret_set(RBC_MESH_GPREGRET_CODE_FORCED_REBOOT);
-            sd_nvic_SystemReset();
+#if (NORDIC_SDK_VERSION >= 11)
+            NVIC_SystemReset();
+#else
+            sd_nvic_SystemReset(); // can't be found >11?
```
Comment (trond-snekvik) [inline nRF51/rbc_mesh/src/mesh_aci.c]: It's been moved to nrf_nvic.h for S130v2

File: nRF51/rbc_mesh/src/timer.c [hunk 2]
```diff
@@ -99,8 +103,11 @@ void timer_event_handler(void)
         if (NRF_TIMER0->EVENTS_COMPARE[i])
         {
             NRF_TIMER0->EVENTS_COMPARE[i] = 0;
+#ifdef RACE_CONDITION_SOLVED
```
Comment (trond-snekvik) [inline nRF51/rbc_mesh/src/timer.c:106]: Should this be reported as an issue? We've had some trouble here, and hope to change it next release. Have you had any bugs related to this?

File: nRF51/rbc_mesh/src/mesh_aci.c (hunk 1 above)
Comment (mrquincle) [reply nRF51/rbc_mesh/src/mesh_aci.c]: Cool, I adjusted it by adding the include.

File: nRF51/rbc_mesh/src/timer.c (hunk 2 above)
Comment (mrquincle) [reply nRF51/rbc_mesh/src/timer.c:106]: We've several other issues as well, but I first wanted to address the obvious changes. I didn't change the name of the nRF51 folder, so it's easier to merge.

**Comment by mrquincle** [comment]
Is there anything that's holding up this merge and for which we can help out?

**Comment by trond-snekvik** [comment]
We're hoping to add bootloader support for 52 while we're at it, so the process is taking a little longer than expected, but we hope we can get it merged sometime next week :)

**Pull request merged**